src/main/java/org/spongepowered/common/data/processor/data/entity/FlyingDataProcessor.java

```diff
@@ -0,0 +1,140 @@
+import org.spongepowered.api.data.DataContainer;
+import org.spongepowered.api.data.DataHolder;
+import org.spongepowered.api.data.DataTransactionBuilder;
+import org.spongepowered.api.data.DataTransactionResult;
+import org.spongepowered.api.data.key.Key;
+import org.spongepowered.api.data.key.Keys;
+import org.spongepowered.api.data.manipulator.immutable.entity.ImmutableFlyingData;
+import org.spongepowered.api.data.manipulator.mutable.entity.FlyingData;
+import org.spongepowered.api.data.merge.MergeFunction;
+import org.spongepowered.api.data.value.BaseValue;
+import org.spongepowered.common.data.ImmutableDataCachingUtil;
+import org.spongepowered.common.data.manipulator.immutable.entity.ImmutableSpongeFlyingData;
+import org.spongepowered.common.data.manipulator.mutable.entity.SpongeFlyingData;
+import org.spongepowered.common.data.processor.common.AbstractSpongeDataProcessor;
+import org.spongepowered.common.data.value.immutable.ImmutableSpongeValue;
+
+public class FlyingDataProcessor extends AbstractSpongeDataProcessor<FlyingData, ImmutableFlyingData> {
+
+    @Override
+    public boolean supports(DataHolder dataHolder) {
+        return dataHolder instanceof net.minecraft.entity.Entity;
+    }
+
+    @SuppressWarnings("unused")
+    @Override
+    public Optional<FlyingData> from(DataHolder dataHolder) {
+        if (supports(dataHolder)) {
+            boolean flying = ((net.minecraft.entity.Entity) dataHolder).isAirBorne;
+            if (dataHolder instanceof EntityPlayer) {
+                EntityPlayer player = (EntityPlayer) dataHolder;
+                flying = player.capabilities.isFlying;
+            }
+            final SpongeFlyingData flyingData = new SpongeFlyingData(flying);
+            return Optional.<FlyingData>of(flyingData);
+        } else {
+            return Optional.absent();
+        }
+
+    }
+
+    @Override
+    public Optional<FlyingData> fill(DataHolder dataHolder, FlyingData manipulator, MergeFunction overlap) {
+        if (supports(dataHolder)) {
+            final FlyingData merged =
+                    checkNotNull(overlap, "MergeFunction cannot be null!").merge(checkNotNull(manipulator).copy(), from(dataHolder).get());
+            manipulator.set(Keys.IS_FLYING, merged.flying().get());
+            return Optional.of(manipulator);
+        }
+        return Optional.absent();
+    }
+
+    @Override
+    public Optional<FlyingData> fill(DataContainer container, FlyingData flyingData) {
+        flyingData.set(Keys.IS_FLYING, getData(container, Keys.IS_FLYING));
+        return Optional.of(flyingData);
+    }
+
+    @Override
+    public DataTransactionResult set(DataHolder dataHolder, FlyingData manipulator, MergeFunction function) {
+        if (this.supports(dataHolder)) {
+            if (dataHolder instanceof EntityPlayer) {
+                EntityPlayer player = (EntityPlayer) dataHolder;
+                player.capabilities.isFlying = manipulator.get(Keys.IS_FLYING).get();
```

Comment:
Instead of repeatedly calling `manipulator.get(Keys.IS_FLYING).get()`, assign it to a local variable. 
Likewise, you'll want to assign an `ImmutableValue<Boolean>` for the previous flying state value.